src/modules/UI/Wallets/action.js

```diff
@@ -161,16 +162,23 @@ export const editCustomToken = (walletId: string, tokenObj: any, oldCurrencyCode
           dispatch(updateExistingTokenSuccess(tokenObj))
           Actions.pop()
         })
-        .catch((e) => dispatch(displayErrorAlert(e)))
+        .catch((e) => {
+          dispatch(displayErrorAlert(e.message))
+          dispatch(editCustomTokenFailure())
+        })
       } else { // replacing an existing but invisible token CASE 3
         addTokenAsync(walletId, tokenObj, state) // update the receiving token
         .then(() => {
           deleteCustomTokenAsync(walletId, oldCurrencyCode, getState) // delete the sending token
           .then((coreWalletsToUpdate) => {
             dispatch(overwriteThenDeleteTokenSuccess(tokenObj, oldCurrencyCode, coreWalletsToUpdate))
-            Actions.pop()
+            Actions.walletList()
```

Comment:
Why are you not using pop()? If the edit token scene was pushed onto the stack, you need to use pop() or you'll add more scenes to the stack.